app/models/dunning_campaign_threshold.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class DunningCampaignThreshold < ApplicationRecord
+  include Currencies
+  include PaperTrailTraceable
+
+  belongs_to :dunning_campaign
+
+  validates :amount_cents, numericality: {greater_than_or_equal_to: 0}
+  validates :currency, inclusion: {in: currency_list}
+  validates :currency, uniqueness: {scope: :dunning_campaign_id}
+  validates :dunning_campaign_id, presence: true
```

Comment:
```suggestion
  validates :currency, :amount_cents, presence: true
```

rails by default already validates presence of `dunning_campaign_id` because it's a belongs_to association without the optional flag.

Reply:
true 👍 - same for `currency` and `amount_cents` with `numericality` and `uniqueness` without `allow_nil`defined :) 

removing the line 👍